client/src/+assets/details.tsx

```diff
@@ -130,15 +145,21 @@ export const AssetDetails: NextPage = () => {
             </UiLayout>
 
             <UiDivider />
-
-            <UiButton cover onClick={(e: any) => {
-              openPopup(e)
-              addtoCart()
-            }}>Add to cart</UiButton>
-            {/* <UiButton cover onClick={addtoCart}>Add to cart</UiButton> */}
+            
+            {ownAsset ? 
+              <UiText className={b('already-purchased')}>
+                You already purchased this dataset, <span className={b('already-purchased-link')}><Link href='/profile'>see in your bundle</Link></span>  
+              </UiText> :
+              <UiButton cover onClick={(e: any) => {
+                if(!isConnected) {
+                  loginMetamask()
+                  return
+                }
+                openPopup(e)
+                addtoCart()
+              }}>{isConnected ? 'Purchase' : 'Connect Wallet'}</UiButton>}
```

Comment:
We don't need the user is connected to add something to the car, only in the checkout process we need to check that

Reply:
The issue come when we have already the item then when we connect in the checkout we still can purchase it when shouldn't be allow